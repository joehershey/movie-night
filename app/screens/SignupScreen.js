import React from "react";
import {
  ImageBackground,
  StyleSheet,
  View,
  Text,
  Image,
  TouchableWithoutFeedback,
  TextInput,
  Alert,
  Keyboard,
  Platform,
} from "react-native";

import { COLORS, STYLES } from "../assets/saved";
import { FontAwesome5 } from "@expo/vector-icons";
import fetch from "cross-fetch";

function SignupScreen(props) {
  const [username, changeUsername] = React.useState("");
  const [email, changeEmail] = React.useState("");
  const [password, changePassword] = React.useState("");

  //TODO replace spike code
  function signupAPI(user, pass, setEmail) {
    fetch(props.url + "signup", {
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json; charset=utf-8",
      },
      method: "POST",
      body: JSON.stringify({
        email: setEmail,
        username: user,
        password: pass,
      }),
    })
      .then((response) => response.json())
      .then((responseJson) => {
        console.log(responseJson);
        if (responseJson.user_id === undefined) {
          alert("Unable to sign up");
        } else {
          props.setUser(responseJson.user_id, user, responseJson.token);
          props.navigation.navigate("Landing");
        }
      })
      .catch((error) => {
        console.error(error);
      });
  }

  function testOrAPI(user, pass, setEmail) {
    //signupAPITest(user, pass, setEmail);
    signupAPI(user, pass, setEmail);
  }

  //test input without api
  function signupAPITest(user, pass, email) {
    console.log(user + ":" + pass + ":" + setEmail);
    props.setUser(2, user); //sets user_id as 2, username as input for username
    props.navigation.navigate("Landing");
  }

  const keyboardControl = () => {
    if (Platform.OS == "web") {
    } else {
      Keyboard.dismiss();
    }
  };

  return (
    <View style={styles.background}>
      <TouchableWithoutFeedback onPress={keyboardControl}>
        <View style={styles.content}>
          <TextInput
            style={STYLES.input}
            onChangeText={changeUsername}
            value={username}
            placeholder="Enter username:"
          />

          <TextInput
            style={STYLES.input}
            onChangeText={changeEmail}
            value={email}
            placeholder="Enter email:"
            testID="EnterEmail"
          />

          <TextInput
            style={STYLES.input}
            onChangeText={changePassword}
            value={password}
            placeholder="Enter password:"
            secureTextEntry={true}
            testID="EnterPassword"
          />

          <TouchableWithoutFeedback testID="SignUpButton"
            onPress={() => {
              if (username == "" || password == "" || email == "") {
                Alert.alert("Please fill out all fields");
              } else {
                testOrAPI(username, password, email);
              }
            }}
          >
            <View style={[STYLES.lgButton, STYLES.btn]}>
              <Text style={[{ color: "white", fontSize: 30 }]}>Sign Up</Text>
            </View>
          </TouchableWithoutFeedback>

          <TouchableWithoutFeedback testID="NavigateButton"
            onPress={() => props.navigation.navigate("Welcome")}
          >
            <View style={[STYLES.returnBtn]}>
              <FontAwesome5
                name="arrow-left"
                color={COLORS.primary}
                size={30}
              />
            </View>
          </TouchableWithoutFeedback>
        </View>
      </TouchableWithoutFeedback>
      <View style={styles.spacer}></View>
    </View>
  );
}

const styles = StyleSheet.create({
  background: {
    flex: 1,
    justifyContent: "center",
    alignItems: "stretch", //stretch allows content to fill screen
    backgroundColor: COLORS.primary,
  },
  content: {
    //used with spacer to push content up the screen
    flex: 11,
    justifyContent: "center",
    alignItems: "center",
  },
  logo: {
    //to be added
    width: 275,
    height: 200,
    margin: 30,
  },
});

export default SignupScreen;
