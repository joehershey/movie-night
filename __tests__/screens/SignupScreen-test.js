import SignupScreen from "../../app/screens/SignupScreen";
import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { TouchableWithoutFeedback, TextInput } from "react-native";
import { shallow, configure } from 'enzyme';
import {render, fireEvent} from '@testing-library/react-native';

configure({adapter: new Adapter()});

describe('Test Signup Screen', () => {
    beforeEach( async () => {
        global.tree = renderer.create(<SignupScreen />);
    });

    it("Signup screen renders properly", async () => {
        await expect(global.tree).toMatchSnapshot();
    });

    it("Does basic test", () => {
        const navigate = jest.fn()
        props = {
            navigation: {
                navigate
            }
        }
        const { getByTestId } = render(<SignupScreen {...props}/>);
        fireEvent.changeText(getByTestId('EnterEmail'), 'JESTEmail');
        fireEvent.changeText(getByTestId('EnterPassword'), 'JESTpassword');
        fireEvent.press(getByTestId('SignUpButton'));
        fireEvent.press(getByTestId('NavigateButton'));
        // expect(getByTestId('MobileCurrentPassword').props.value).toEqual('JESTTestPassword');
        expect(navigate).toBeCalledTimes(1); // confirms navigate call
    });

    it("Check keyboard control on press", () => {
        const container = shallow(<SignupScreen />);
        const keyboardControl = container.find(TouchableWithoutFeedback).at(0); // finds onPresss => keyboardControl()
        keyboardControl.props().onPress(); // calls keyboardControl()
        expect(keyboardControl.exists());
    })

    it("Simulate sign up button on press", () => {
        const container = shallow(<SignupScreen />);
        const signUp = container.find(TouchableWithoutFeedback).at(1); // finds onPress => Alert else testOrAPI()
        signUp.props().onPress(); // calls Alert b/c username, password, email are blank
        expect(signUp.exists());
    })

    it("Check welcome screen navigation on sign up press", () => {
        const navigate = jest.fn();
        const props = {
            navigation: {
                navigate
            }
        }
        const container = shallow(<SignupScreen {...props} />);
        const welcome = container.find(TouchableWithoutFeedback).last(); // finds onPress => props.navigation.navigate()
        welcome.props().onPress(); // calls props.navigation.navigate()
        expect(welcome.exists());
        expect(navigate.mock.calls[0][0]).toBe('Welcome'); // navigation navigates to the welcome screen
    })

});