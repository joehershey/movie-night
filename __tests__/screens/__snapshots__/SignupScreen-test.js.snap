// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Signup Screen Signup screen renders properly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#1d1f1e",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 11,
        "justifyContent": "center",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Enter username:"
      rejectResponderTermination={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 0,
          "height": 50,
          "justifyContent": "center",
          "margin": 4,
          "padding": 10,
          "width": "60%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Enter email:"
      rejectResponderTermination={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 0,
          "height": 50,
          "justifyContent": "center",
          "margin": 4,
          "padding": 10,
          "width": "60%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Enter password:"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 0,
          "height": 50,
          "justifyContent": "center",
          "margin": 4,
          "padding": 10,
          "width": "60%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#166fdb",
            "height": 70,
            "top": 30,
            "width": "50%",
          },
          Object {
            "alignItems": "center",
            "borderRadius": 20,
            "justifyContent": "center",
            "margin": 5,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 30,
            },
          ]
        }
      >
        Sign Up
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderRadius": 10,
            "left": 20,
            "margin": 10,
            "padding": 10,
            "position": "absolute",
            "top": 50,
          },
        ]
      }
    >
      <Text />
    </View>
  </View>
  <View />
</View>
`;
